app/gui2/src/components/TextFormattingSelector.vue

```diff
@@ -0,0 +1,132 @@
+<script setup lang="ts">
+import icons from '@/assets/icons.svg'
+import DropdownMenu from '@/components/DropdownMenu.vue'
+import SvgButton from '@/components/SvgButton.vue'
+import { ref } from 'vue'
+import { TextFormatOptions } from './visualizations/TableVisualization.vue'
+
+const emit = defineEmits<{
+  changeFormat: [formatValue: TextFormatOptions]
+}>()
+
+const textFormatterSelected = ref(TextFormatOptions.Partial)
+const setTextFormatterSelected = (option: TextFormatOptions) => {
+  textFormatterSelected.value = option
+  emit('changeFormat', textFormatterSelected.value)
+}
+
+const iconPath = `${icons}#paragraph`
+
+const open = ref(false)
+
+const toggleMenu = () => {
+  open.value = !open.value
+}
+</script>
+
+<template>
+  <DropdownMenu v-model:open="open" class="TextFormattingSelector">
+    <template #button
+      ><SvgButton name="paragraph" title="Text Display Options" @click.stop="toggleMenu()" />
+    </template>
+
+    <template #entries>
+      <button
+        :class="`${textFormatterSelected === TextFormatOptions.On && 'selected'}`"
+        :onclick="() => setTextFormatterSelected(TextFormatOptions.On)"
+        title="Text displayed in monaspace font and all whitespace characters displayed as a symbol"
+      >
+        <svg viewBox="0 0 16 16" width="16" height="16" style="stroke: black; fill: #000000">
+          <use :href="`${iconPath}`" />
+        </svg>
+        <div>All Whitespace rendering</div>
+      </button>
+
+      <button
+        :onclick="() => setTextFormatterSelected(TextFormatOptions.Partial)"
+        :class="`${textFormatterSelected === TextFormatOptions.Partial && 'selected'}`"
+        title="Text displayed in monaspace font, only multiple spaces displayed with &#183;"
+      >
+        <svg viewBox="0 0 16 16" width="16" height="16" style="stroke: grey; fill: #808080">
+          <use :href="`${iconPath}`" />
+        </svg>
+        <div>Partial whitespace rendering</div>
+      </button>
+
+      <button
+        :onclick="() => setTextFormatterSelected(TextFormatOptions.Off)"
+        :class="`${textFormatterSelected === TextFormatOptions.Off && 'selected'}`"
+        title="No formatting applied to text"
+      >
+        <div class="strikethrough">
+          <svg viewBox="0 0 16 16" width="16" height="16">
+            <use :href="`${iconPath}`" />
+          </svg>
+        </div>
+
+        <div>No whitespace rendering</div>
+      </button>
+    </template>
+  </DropdownMenu>
+</template>
+
+<style scoped>
+.TextFormattingSelector {
+  background: var(--color-frame-bg);
+  border-radius: 16px;
+}
+
+:deep(.DropdownMenuContent) {
+  margin-top: 2px;
+  padding: 4px;
+
+  > * {
+    display: flex;
+    align-items: center;
+    padding-left: 8px;
+    padding-right: 8px;
+  }
+}
+
+button {
```

Comment:
Matching by element in a scoped style should be avoided: https://vuejs.org/api/sfc-css-features.html#scoped-style-tips